File: app/src/main/java/com/unip/aps/aps_noticias/ui/activity/SplashActivity.java
package com.unip.aps.aps_noticias.ui.activity;

import android.os.Bundle;
import android.os.Handler;

import com.unip.aps.aps_noticias.R;

public class SplashActivity extends BaseActivity {

    private long TIME_DELAY = 3000;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        Handler handler = new Handler();
        handler.postDelayed(new Runnable() {
            @Override
            public void run() {
                openActivity(SplashActivity.this, LoginActivity.class);
                finish();
            }
        }, TIME_DELAY);
    }
}

File: app/src/main/java/com/noticias_now/splash/SplashActivity.kt
package com.noticias_now.splash

import android.content.Intent
import android.os.Bundle
import androidx.lifecycle.Observer
import br.com.ricarlo.common.ui.base.BaseActivity
import br.com.ricarlo.common.util.ViewState
import br.com.ricarlo.common.util.extensions.doOnTransitionCompleted
import com.noticias_now.R
import com.noticias_now.databinding.ActivitySplashBinding
import com.noticias_now.home.HomeActivity
import com.noticias_now.login.LoginActivity
import org.koin.android.viewmodel.ext.android.viewModel

class SplashActivity : BaseActivity<ActivitySplashBinding>() {

    private val viewModel: SplashViewModel by viewModel()

    override fun getLayoutRes(): Int {
        return R.layout.activity_splash
    }

    override fun initView(savedInstanceState: Bundle?) {

        binding.splashMotionLayout.doOnTransitionCompleted { _, _ ->
            viewModel.setCompletedAnimation()
        }
        binding.splashMotionLayout.startLayoutAnimation()

        subscribeUI()
    }

    private fun subscribeUI() {
        viewModel.result.observe(
            this,
            Observer {
                when (it) {
                    is ViewState.Loading -> {
                    }
                    is ViewState.Success -> {
                        val cls = when (it.data) {
                            is SplashEvent.LaunchLogin -> {
                                LoginActivity::class.java
                            }
                            is SplashEvent.LaunchHome -> {
                                HomeActivity::class.java
                            }
                        }
                        startActivity(Intent(this, cls))
                        finish()
                    }
                    is ViewState.Error -> {
                        handlerError(it.error)
                    }
                }
            }
        )
    }
}
